lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -426,12 +427,12 @@ def pass_hook_checks(self, event):
             event.defer()
             return False
 
-        if not self.is_ca_compatible():
+        # race condition where mongos cannot start without TLS certificates, but mongos cannot
+        # request TLS certificates without knowing the name of the config-server.
+        if self.is_waiting_to_request_certs():
```

Comment:
We do lose the check on ca compatibility here however.
Shouldn't we do something like  this?
```
if self.is_waiting_torequest_certs:
…
if not self.is_compatible():
…
```

Reply:
Good eye @Gu1nness 

Its actually not useful here, infact **it is harmful**. It blocks the mongos router from requesting its certs (until it requests the certs it won't know the CA) 

This check is done elsewhere in the charm to ensure that mongos reports on a mismatched CA. So the user won't lose this context :)

Reply:
Ok, thanks :)